app/controllers/v1/ControllerUtils.scala

```diff
@@ -103,38 +107,62 @@ trait ControllerUtils extends Controller with StrictLogging with Properties with
     case ex => InternalServerError(errAsJson(ex.toString, ex.getCause.toString))
   }
 
+  protected def louSearch(baseUrl: Uri, id: String, period: String)(implicit ws: RequestGenerator): Future[Result] = id match {
+    case validId if (id.length >= MINIMUM_KEY_LENGTH) => {
+      LOGGER.info(s"Sending request to ${baseUrl.toString} to retrieve Unit Links")
+      ws.singleGETRequest(baseUrl.toString) map {
+        case response if response.status == OK => {
+          val unitResp = response.json.as[StatisticalUnitLinkType]
+          val mapOfRecordKeys = Map(LOU.toString -> (unitResp \ "id").as[String])
+          val period = (unitResp \ "period").as[String]
+          val respRecords = parsedRequest(mapOfRecordKeys, unitResp, Some(period))
+          val json = (Seq(unitResp) zip respRecords).map(x => toJson(x, LOU.toString, period)).head
+          Ok(json).as(JSON)
+        }
+        case response if response.status == NOT_FOUND => NotFound(response.body).as(JSON)
+      }
+    }
+    case _ => BadRequest(Messages("controller.invalid.id", id, MINIMUM_KEY_LENGTH)).future
+  }
```

Comment:
I don't see any recovery on your ws.singleGETRequest.  If you encounter some kind of connection issue, your resulting Future will be a Failure, and your map will not do anything.  You need to recover the Failure so that the controller returns a Successful future containing a 500 status code.